Client/multiplayer_sa/CMultiplayerSA.cpp

```diff
@@ -1,6872 +1,6917 @@
-    00748227   FF15 9C828500    CALL DWORD PTR DS:[<&USER32.SetCursor>]  ; USER32.SetCursor
-    */
-    BYTE originalCode[16] = {0x6A, 0x00, 0xFF, 0x15, 0xEC, 0x82, 0x85, 0x00, 0x6A, 0x00, 0xFF, 0x15, 0x9C, 0x82, 0x85, 0x00};
-
-    if (bAllow)
-    {
-        MemSet((LPVOID)ADDR_CursorHiding, 0x90, 16);
-    }
-    else
-    {
-        MemCpy((LPVOID)ADDR_CursorHiding, &originalCode, 16);
-    }
-}
-
-CShotSyncData* CMultiplayerSA::GetLocalShotSyncData()
-{
-    return GetLocalPedShotSyncData();
-}
-
-void CMultiplayerSA::DisablePadHandler(bool bDisabled)
-{
-    // DISABLE GAMEPADS (testing)
-    if (bDisabled)
-        MemPut<BYTE>(0x7449F0, 0xC3);
-    else
-        MemPut<BYTE>(0x7449F0, 0x8B);
-}
-
-void CMultiplayerSA::GetHeatHaze(SHeatHazeSettings& settings)
-{
-    settings.ucIntensity = *(int*)0x8D50E8;
-    settings.ucRandomShift = *(int*)0xC402C0;
-    settings.usSpeedMin = *(int*)0x8D50EC;
-    settings.usSpeedMax = *(int*)0x8D50F0;
-    settings.sScanSizeX = *(int*)0xC40304;
-    settings.sScanSizeY = *(int*)0xC40308;
-    settings.usRenderSizeX = *(int*)0xC4030C;
-    settings.usRenderSizeY = *(int*)0xC40310;
-    settings.bInsideBuilding = *(bool*)0xC402BA;
-}
-
-void DoSetHeatHazePokes(const SHeatHazeSettings& settings, int iHourStart, int iHourEnd, float fFadeSpeed, float fInsideBuildingFadeSpeed,
-                        bool bAllowAutoTypeChange)
-{
-    MemPutFast<int>(0x8D50D4, iHourStart);
-    MemPutFast<int>(0x8D50D8, iHourEnd);
-
-    MemPutFast<float>(0x8D50DC, fFadeSpeed);
-    MemPutFast<float>(0x8D50E0, fInsideBuildingFadeSpeed);
-
-    MemPutFast<int>(0x8D50E8, settings.ucIntensity);
-    MemPutFast<int>(0xC402C0, settings.ucRandomShift);
-    MemPutFast<int>(0x8D50EC, settings.usSpeedMin);
-    MemPutFast<int>(0x8D50F0, settings.usSpeedMax);
-    MemPutFast<int>(0xC40304, settings.sScanSizeX);
-    MemPutFast<int>(0xC40308, settings.sScanSizeY);
-    MemPutFast<int>(0xC4030C, settings.usRenderSizeX);
-    MemPutFast<int>(0xC40310, settings.usRenderSizeY);
-    MemPutFast<bool>(0xC402BA, settings.bInsideBuilding);
-
-    if (bAllowAutoTypeChange)
-        MemPut<BYTE>(0x701455, 0x83);            // sub
-    else
-        MemPut<BYTE>(0x701455, 0xC3);            // retn
-}
-
-void CMultiplayerSA::SetHeatHaze(const SHeatHazeSettings& settings)
-{
-    if (settings.ucIntensity != 0)
-        DoSetHeatHazePokes(settings, 0, 24, 1.0f, 1.0f, false);            // 24 hrs
-    else
-        DoSetHeatHazePokes(settings, 38, 39, 1.0f, 1.0f, false);            // 0 hrs
-
-    m_bHeatHazeCustomized = true;
-    ApplyHeatHazeEnabled();
-}
-
-void CMultiplayerSA::ResetHeatHaze(void)
-{
-    SHeatHazeSettings settings;
-    settings.ucIntensity = 0x50;
-    settings.ucRandomShift = 0x0;
-    settings.usSpeedMin = 0x0C;
-    settings.usSpeedMax = 0x12;
-    settings.sScanSizeX = 0x4B;
-    settings.sScanSizeY = 0x50;
-    settings.usRenderSizeX = 0x50;
-    settings.usRenderSizeY = 0x55;
-    se
```

Comment:
why do you have two asm blocks? you can simply put this code after popad in first asm block.